domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/casefield/CaseFieldEntityACLValidatorImpl.java

```diff
@@ -14,12 +21,15 @@ public ValidationResult validate(final CaseFieldEntity caseField,
                                      final CaseFieldEntityValidationContext caseFieldEntityValidationContext) {
 
         final ValidationResult validationResult = new ValidationResult();
-
+        Set<String> caseRoles = caseFieldEntityValidationContext.getCaseRoles().stream().map(CaseRoleEntity::getReference).collect(Collectors.toSet());
         for (CaseFieldACLEntity entity : caseField.getCaseFieldACLEntities()) {
 
             if (null == entity.getUserRole()) {
                 validationResult.addError(new CaseFieldEntityInvalidUserRoleValidationError(entity,
                     new AuthorisationCaseFieldValidationContext(caseField, caseFieldEntityValidationContext)));
+            } else if (isCaseRole(entity.getUserRole().getReference()) && !caseRoles.contains(entity.getUserRole().getReference())) {
+                validationResult.addError(new CaseFieldEntityInvalidUserRoleValidationError(entity,
```

Comment:
beware we are throwing an `CaseFieldEntityInvalidUserRoleValidationError` here. So the error message would be: 
`Invalid UserRole for case type '%s', case field '%s'"` which I'm sure will confuse people. Because case roles have not the same syntax of user roles. 
We should create a new error class, with a dedicated error message. Error reporting is very important. It should say something like:
`Unknown case role <x>`. Please make sure <x> is declared in the list of supported case roles for the case type`